filament/src/MaterialDefinition.cpp

```diff
@@ -0,0 +1,402 @@
+#include "MaterialDefinition.h"
+
+#include "Froxelizer.h"
+#include "MaterialParser.h"
+
+#include <ds/ColorPassDescriptorSet.h>
+
+#include <details/Engine.h>
+
+#include <utils/Logger.h>
+#include <utils/Panic.h>
+
+namespace filament {
+
+using namespace backend;
+using namespace utils;
+
+namespace {
+
+static const char* toString(ShaderModel model) {
+    switch (model) {
+        case ShaderModel::MOBILE:
+            return "mobile";
+        case ShaderModel::DESKTOP:
+            return "desktop";
+    }
+}
```

Comment:
move this to `DriverEnums.h`, rename to `to_string` and return a `std::string_view` (similar to other types in `DriverEnums.h`)